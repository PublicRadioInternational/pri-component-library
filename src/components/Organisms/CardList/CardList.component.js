/**
 * @file CardList.component.js
 * Exports a card list component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './CardList.css';

import Section from '../Content/Section.component';

/**
 * Component that renders a Card List.
 */
const CardList = ({ category, children, url, src, categoryDescription }) => (
  <Section className={`${styles.list} ${styles[category]}`}>
    <header className={styles.header}>
      <a href={url} className={styles.logoLink}>
        {src && <img src={src} alt="" className={styles.logo} />}
        <span>{categoryDescription}</span>
      </a>
    </header>
    {children}
  </Section>
);

CardList.propTypes = {
  category: PropTypes.string,
  children: PropTypes.node,
  url: PropTypes.string,
  src: PropTypes.string,
  categoryDescription: PropTypes.string
};

CardList.defaultProps = {
  category: 'world',
  children: [],
  url: null,
  src: null,
  categoryDescription: null
};

export default CardList;
