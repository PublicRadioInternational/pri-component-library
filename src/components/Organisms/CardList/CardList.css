/* import colors... */
@value colors: "../../00_global/colors.css";
@value grayLighter, blue, world, americaAbroad, globalPost, liveWire, livingEarth, otherhood, playdate, scienceHappiness, sciFri, studio, takeaway, thingsBoom, undiscovered, acrossWomensLives, globalNation, globalSecurity, livablePlanet from colors;

/* import fonts... */
@value fonts: "../../00_global/fonts.css";
@value heading from fonts;

.list {
  background: white;
  border-top: 3px solid #ddd;
  border-radius: 0.25rem;
  word-wrap: break-word;
  margin-bottom: 1rem;
}

.list:last-child {
  margin-bottom: 0;
}

.header {
  border-bottom: 1px solid grayLighter;
  padding: 0.5rem 1rem 0.7rem;
  font-family: heading;
  font-weight: bold;
  line-height: 50px;
}

.logoLink {
  align-items: center;
  text-decoration: none;
}

.logo {
  float: left;
  width: 50px;
  height: 50px;
  margin-right: 10px;
}

.america-abroad {
  border-top-color: americaAbroad;
}

.boston-calling {
  border-top-color: bostonCalling;
}

.globalpost {
  border-top-color: globalPost;
}

.live-wire {
  border-top-color: liveWire;
}

.living-on-earth {
  border-top-color: livingEarth;
}

.otherhood {
  border-top-color: otherhood;
}

.playdate {
  border-top-color: playdate;
}

.pris-the-world {
  border-top-color: world;
}

.science-of-happiness {
  border-top-color: scienceHappiness;
}

.science-friday {
  border-top-color: sciFri;
}

.studio-360 {
  border-top-color: studio;
}

.the-takeaway {
  border-top-color: takeaway;
}

.the-world-in-words {
  border-top-color: worldWords;
}

.things-that-go-boom {
  border-top-color: thingsBoom;
}

.undiscovered {
  border-top-color: undiscovered;
}

.across-womens-lives {
  border-top-color: acrossWomensLives;
}

.global-nation {
  border-top-color: globalNation;
}

.global-security {
  border-top-color: globalSecurity;
}

.livable-planet {
  border-top-color: livablePlanet;
}
