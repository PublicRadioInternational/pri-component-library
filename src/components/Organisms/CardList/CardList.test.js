/**
 * @file CardList.test.js
 * Contains tests for CardList.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import CardList from './CardList.component';

describe('<CardList />', () => {
  it('Matches the Card List snapshot', () => {
    const component = renderer
      .create(
        <CardList
          src="https://interactive-dev.pri.org/staging/prototypes/homepage/themes/pri/images/logo_tw.png"
          category="world"
          url="#"
          categoryDescription="PRIs The World"
        >
          <span>One</span>
          <span>Two</span>
        </CardList>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
