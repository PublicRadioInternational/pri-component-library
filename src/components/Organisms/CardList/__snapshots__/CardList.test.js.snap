// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> Matches the Card List snapshot 1`] = `
<section
  className="className && list world"
>
  <header
    className="header"
  >
    <a
      className="logoLink"
      href="#"
    >
      <img
        alt="PRIs The World"
        className="logo"
        src="https://interactive-dev.pri.org/staging/prototypes/homepage/themes/pri/images/logo_tw.png"
      />
      <span>
        PRIs The World
      </span>
    </a>
  </header>
  <span>
    One
  </span>
  <span>
    Two
  </span>
</section>
`;
